toolchain/check/handle_binding_pattern.cpp

```diff
@@ -127,7 +127,7 @@ static auto HandleAnyBindingPattern(Context& context, Parse::NodeId node_id,
       // add a proper diagnostic.
       context.TODO(node_id, "_ used as field name");
     }
-    cast_type_id = AsConcreteType(
+    RequireConcreteType(
```

Comment:
If this returns false (and produced a diagnotic), should we set `cast_type_inst_id = ErrorInst`, so that we can avoid further diagnostics, and so that we don't place a non-concrete type (inst) in places that we require a concrete one?

Reply:
I touched on this in the commit message: I don't think it's problematic to have a field type that isn't a concrete type (for example, the base type may not be concrete, and that's OK), so while we do want to reject this, it seems like it's OK to preserve the type in the error case. But I don't feel strongly about this, and maybe it's better to maintain a stronger invariant here. Sure, I'll do that.

Reply:
Done.